packages/editor/src/components/rich-text/style.scss

```diff
@@ -65,8 +55,27 @@
 				border-radius: 2px;
 				box-shadow: 0 0 0 1px $light-gray-400;
 				background: $light-gray-400;
+
+				// Enforce a dark text color so active inline boundaries
+				// are always readable.
+				// See https://github.com/WordPress/gutenberg/issues/9508
+				color: $dark-gray-900;
 			}
 		}
+
+		// Link inline boundaries get special colors.
+		a[data-mce-selected] {
+			box-shadow: 0 0 0 1px $blue-medium-100;
+			background: $blue-medium-100;
```

Comment:
Is there sufficient colour contrast here? To my eyes it's fine, but then I have good eyesight and am right now using my 5K monitor 🤷‍♂️

Reply:
Good catch; updated this to `$blue-medium-900` to achieve WCAG AA compatibility.